addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/handler/OneWireGPIOHandler.java

```diff
@@ -0,0 +1,155 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Invalid path to w1_slave file");
+            return;
+        }
+
+        refreshTime = ((Number) configuration.get(REFRESH_TIME)).intValue();
+        if (refreshTime.intValue() <= 0) {
+            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}.", refreshTime,
+                    DEFAULT_REFRESH_TIME);
+            refreshTime = DEFAULT_REFRESH_TIME;
+        }
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable refresher = new Runnable() {
+            @Override
+            public void run() {
+                List<Channel> channels = getThing().getChannels();
+                for (Channel channel : channels) {
+                    publishSensorValue(channel.getUID());
+                }
+            }
+        };
+
+        scheduler.scheduleWithFixedDelay(refresher, 0, refreshTime.intValue(), TimeUnit.SECONDS);
+        logger.debug("Start automatic refresh every {} seconds", refreshTime.intValue());
+    }
+
+    private void publishSensorValue(ChannelUID channelUID) {
+        String channelID = channelUID.getId();
+        State state = null;
+        switch (channelID) {
+            case TEMPERATURE:
+                BigDecimal temp = readSensorTemperature(gpioBusFile);
+                temp = temp.setScale(2, BigDecimal.ROUND_HALF_UP);
+                state = new DecimalType(temp);
+                break;
+        }
+        if (state != null) {
+            updateState(channelID, state);
+        } else {
+            logger.debug("Can not update channel with ID : {} - channel name might be wrong!", channelID);
+        }
+    }
+
+    private BigDecimal readSensorTemperature(String gpioFile) {
+        try (Stream<String> stream = Files.lines(Paths.get(gpioFile))) {
+            Optional<String> temperatureLine = stream
+                    .filter(s -> s.contains(OneWireGPIOBindingConstants.FILE_TEMP_MARKER)).findFirst();
+            if (temperatureLine.isPresent()) {
+                String line = temperatureLine.get();
+                String tempString = line.substring(line.indexOf(OneWireGPIOBindingConstants.FILE_TEMP_MARKER)
+                        + OneWireGPIOBindingConstants.FILE_TEMP_MARKER.length());
+                Integer intTemp = Integer.parseInt(tempString);
+                if (getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+                return BigDecimal.valueOf(intTemp).movePointLeft(3);
+            } else {
+                logger.debug(
+                        "GPIO file didn't contain line with 't=' where temperature value should be available. Check if configuration points to the proper file");
+                updateStatus(ThingStatus.OFFLINE);
+                return OneWireGPIOBindingConstants.INVALID_TEMPERATURE;
```

Comment:
Instead of an invalid temperature please return null here and do not update the channel state at all. The last read temperature will stay the current state on the channel. See this comment: https://github.com/eclipse/smarthome/issues/4612#issuecomment-354737064